llvm: Build broken with clang 5.0.0

Linking breaks with current clang:

```
$ make TOOLCHAIN=llvm
Building application "$app" for "bluepill" with MCU "stm32f1".

"make" -C /boards/bluepill
"make" -C /core
"make" -C /cpu/stm32f1
"make" -C /drivers
"make" -C /sys
"make" -C /cpu/cortexm_common
"make" -C /sys/auto_init
"make" -C /drivers/periph_common
"make" -C /cpu/cortexm_common/periph
"make" -C /cpu/stm32_common
"make" -C /sys/div
"make" -C /cpu/stm32_common/periph
"make" -C /sys/isrpipe
"make" -C /sys/newlib_syscalls_default
"make" -C /sys/pm_layered
"make" -C /sys/tsrb
"make" -C /sys/uart_stdio
"make" -C /sys/xtimer
"make" -C /cpu/stm32f1/periph
clang-5.0: warning: argument unused during compilation: '-nostartfiles' [-Wunused-command-line-argument]
clang-5.0: warning: argument unused during compilation: '-specs=nano.specs' [-Wunused-command-line-argument]
/usr/bin/ld.lld: error: unable to find library -lm
/usr/bin/ld.lld: error: unable to find library -lgcc
/usr/bin/ld.lld: error: unable to find library -lc
/usr/bin/ld.lld: error: unable to find library -lnosys
/usr/bin/ld.lld: error: cannot open stm32_common.ld: No such file or directory
/usr/bin/ld.lld: error: unable to find library -lc
/usr/bin/ld.lld: error: unable to find library -lm
/usr/bin/ld.lld: error: unable to find library -lclang_rt.builtins-arm.a
clang-5.0: error: ld.lld command failed with exit code 1 (use -v to see invocation)
make: *** [/examples/splitflap/../../Makefile.include:321: link] Error 1
```

--- a/Makefile.include
+++ b/Makefile.include
@@ -310,7 +310,7 @@ LINKFLAGPREFIX ?= -Wl,
 
 DIRS += $(EXTERNAL_MODULE_DIRS)
 
-_LINK = $(if $(CPPMIX),$(CXX),$(LINK)) $(UNDEF) $(LINKFLAGPREFIX)--start-group $(BASELIBS) -lm $(LINKFLAGPREFIX)--end-group  $(LINKFLAGPREFIX)-Map=$(BINDIR)/$(APPLICATION).map $(LINKFLAGS)
+_LINK = $(if $(CPPMIX),$(LINKXX),$(LINK)) $(UNDEF) $(LINKFLAGPREFIX)--start-group $(BASELIBS) -lm $(LINKFLAGPREFIX)--end-group  $(LINKFLAGPREFIX)-Map=$(BINDIR)/$(APPLICATION).map $(LINKFLAGS)
 
 ifeq ($(BUILD_IN_DOCKER),1)
 link: ..in-docker-container
@@ -320,7 +320,7 @@ link: ..compiler-check ..build-message $(RIOTBUILD_CONFIG_HEADER_C) $(USEPKG:%=$
 	$(Q)DIRS="$(DIRS)" "$(MAKE)" -C $(APPDIR) -f $(RIOTMAKE)/application.inc.mk
 ifeq (,$(RIOTNOLINK))
 ifeq ($(BUILDOSXNATIVE),1)
-	$(Q)$(if $(CPPMIX),$(CXX),$(LINK)) $(UNDEF) -o $(ELFFILE) $$(find $(BASELIBS) -size +8c) $(LINKFLAGS) $(LINKFLAGPREFIX)-no_pie
+	$(Q)$(if $(CPPMIX),$(LINKXX),$(LINK)) $(UNDEF) -o $(ELFFILE) $$(find $(BASELIBS) -size +8c) $(LINKFLAGS) $(LINKFLAGPREFIX)-no_pie
 else
 	$(Q)$(_LINK) -o $(ELFFILE)
 endif

--- a/makefiles/toolchain/gnu.inc.mk
+++ b/makefiles/toolchain/gnu.inc.mk
@@ -8,6 +8,7 @@ export AR         = $(PREFIX)ar
 endif
 export AS         = $(PREFIX)as
 export LINK       = $(PREFIX)gcc
+export LINKXX     = $(PREFIX)g++
 export SIZE       = $(PREFIX)size
 export OBJCOPY   ?= $(shell command -v $(PREFIX)objcopy gobjcopy objcopy | head -n 1)
 ifeq ($(OBJCOPY),)

--- a/makefiles/toolchain/llvm.inc.mk
+++ b/makefiles/toolchain/llvm.inc.mk
@@ -9,12 +9,14 @@ endif
 export CC          = clang
 export CXX         = clang++
 export CCAS       ?= $(CC)
-export LINK        = $(CC)
 export AS          = $(LLVMPREFIX)as
 export AR          = $(LLVMPREFIX)ar
 export NM          = $(LLVMPREFIX)nm
-# There is no LLVM linker yet, use GNU binutils.
-#export LINKER      = $(LLVMPREFIX)ld
+# LLVM does have a linker, however, it is not entirely
+# compatible with GCC. For instance spec files as used in
+# `makefiles/libc/newlib.mk` are not supported. Therefore
+# we just use GCC for now.
+export LINK        = $(PREFIX)gcc
 # objcopy does not have a clear substitute in LLVM, use GNU binutils
 #export OBJCOPY     = $(LLVMPREFIX)objcopy
 export OBJCOPY    ?= $(shell command -v $(PREFIX)objcopy gobjcopy objcopy | head -n 1)
@@ -52,7 +54,8 @@ ifneq (,$(TARGET_ARCH))
   # Tell clang to cross compile
   export CFLAGS     += -target $(TARGET_ARCH)
   export CXXFLAGS   += -target $(TARGET_ARCH)
-  export LINKFLAGS  += -target $(TARGET_ARCH)
+  # We currently don't use LLVM for linking (see comment above).
+  #export LINKFLAGS  += -target $(TARGET_ARCH)
 
   # Use the wildcard Makefile function to search for existing directories matching
   # the patterns above. We use the -isystem gcc/clang argument to add the include

--- a/makefiles/toolchain/llvm.inc.mk
+++ b/makefiles/toolchain/llvm.inc.mk
@@ -17,6 +17,7 @@ export NM          = $(LLVMPREFIX)nm
 # `makefiles/libc/newlib.mk` are not supported. Therefore
 # we just use GCC for now.
 export LINK        = $(PREFIX)gcc
+export LINKXX      = $(PREFIX)g++
 # objcopy does not have a clear substitute in LLVM, use GNU binutils
 #export OBJCOPY     = $(LLVMPREFIX)objcopy
 export OBJCOPY    ?= $(shell command -v $(PREFIX)objcopy gobjcopy objcopy | head -n 1)

